pkg/controller/disruption/disruption.go

```diff
@@ -452,9 +478,21 @@ func (dc *DisruptionController) trySync(pdb *policy.PodDisruptionBudget) error {
 		return err
 	}
 
-	currentHealthy := countHealthyPods(pods)
-	err = dc.updatePdbSpec(pdb, currentHealthy, desiredHealthy, expectedCount)
+	currentTime := time.Now()
+	disruptedPods, recheckTime := buildDisruptedPodMap(pods, pdb, currentTime)
```

Comment:
currentTime is observed in the controller manager process while the time the pod was evicted is observed in the api server. If the controller manager runs in a separate machine (is it possible currently?) then you get into clock-skew issues.